### ISSUE TITLE
With namespace set name of kind: Namespace should be augmented as well

### ISSUE THREAD

**Issue by elmariofredo** [kubernetes-sigs/kustomize#2867]
When `namespace` is set it will update all `namespace:` properties but skipping `name:` property of `kind: Namespace`  see example and expected result below.

## Setup example

```
cat <<'EOF' >kustomization.yaml
namespace: nginx-system
resources:
- manifests.yaml
EOF

cat <<'EOF' >manifests.yaml
---
apiVersion: v1
kind: Namespace
metadata:
  name: init-demo
---
apiVersion: v1
kind: Pod
metadata:
  name: init-demo
  namespace: init-demo
spec:
  containers:
  - name: nginx
    image: nginx
EOF
```

## Current output

```
➜ kustomize build
apiVersion: v1
kind: Namespace
metadata:
  name: init-demo
---
apiVersion: v1
kind: Pod
metadata:
  name: init-demo
  namespace: nginx-system
spec:
  containers:
  - image: nginx
    name: nginx
```

## Expected output

```
➜ kustomize build
apiVersion: v1
kind: Namespace
metadata:
  name: nginx-system
---
apiVersion: v1
kind: Pod
metadata:
  name: init-demo
  namespace: nginx-system
spec:
  containers:
  - image: nginx
    name: nginx
```

**Comment by elmariofredo** [comment]
Workaround:

```
namespace: nginx-system

resources:
- manifests.yaml

patchesJson6902:
- target:
    version: v1
    kind: Namespace
    name: init-demo
  patch: |-
    - op: replace
      path: "/metadata/name"
      value: nginx-system
```

**Comment by thande** [comment]
I was also hit by this one, and confirmed it's a regression in behavior as version `{Version:3.5.4 GitCommit:3af514fa9f85430f0c1557c4a0291e62112ab026 BuildDate:2020-01-17T14:23:25+00:00 GoOs:darwin GoArch:amd64}` behaved as expected but after upgrading to the current release `{Version:3.8.2 GitCommit:e2973f6ecc9be6187cfd5ecf5e180f842249b3c6 BuildDate:2020-08-29T19:15:49+01:00 GoOs:darwin GoArch:amd64}` the namespace Kind was no longer set as expected.

**Comment by Shell32-Natsu** [comment]
Closed by #2953 

/close

**Comment by elmariofredo** [comment]
Thanks @easimon 👍
